tests/tests/custom_authority_tests.cpp

```diff
@@ -5267,4 +5834,513 @@ BOOST_AUTO_TEST_CASE(custom_auths) { try {
+            authorize_account_update.auth.weight_threshold = 1;
+            authorize_account_update.enabled = true;
+            authorize_account_update.valid_to = db.head_block_time() + 86400;
+            authorize_account_update.operation_type = operation::tag<account_update_operation>::value;
+
+            // Shall not update the owner key member
+            auto owner_index = member_index<account_update_operation>("owner");
+            restriction no_owner = restriction(owner_index, FUNC(eq), void_t());
+
+            // Shall not update the active key member
+            auto active_index = member_index<account_update_operation>("active");
+            restriction no_active = restriction(active_index, FUNC(eq), void_t());
+
+            // Shall not update the memo key member of the new_options member
+            auto new_options_index = member_index<account_update_operation>("new_options");
+            auto memo_index = member_index<account_options>("memo_key");
+            restriction same_memo = restriction(new_options_index, FUNC(attr),
+                                                vector<restriction>{
+                                                        restriction(memo_index, FUNC(eq), alice.options.memo_key)});
+
+            // Shall not update the extensions member
+            auto ext_index = member_index<account_update_operation>("extensions");
+            restriction no_ext = restriction(ext_index, FUNC(eq), void_t());
+
+            auto owner_special_index = member_index<account_update_operation::ext>("owner_special_authority");
+            restriction no_special_owner = restriction(ext_index, FUNC(attr),
+                                                       vector<restriction>{
+                                                               restriction(owner_special_index, FUNC(eq), void_t())});
+
+            auto active_special_index = member_index<account_update_operation::ext>("active_special_authority");
+            restriction no_special_active = restriction(ext_index, FUNC(attr),
+                                                        vector<restriction>{
+                                                                restriction(active_special_index, FUNC(eq), void_t())});
+
+            // Shall not update the extensions member of the new_options member
+            auto new_options_ext_index = member_index<account_options>("extensions");
+            restriction no_new_options_ext = restriction(new_options_index, FUNC(attr), vector<restriction>{
+                    restriction(new_options_ext_index, FUNC(eq), void_t())});
+
+            // Combine all of the shall not restrictions
+            vector<restriction> shall_not_restrictions = {no_owner, no_active, no_special_owner, no_special_active,
+                                                          same_memo};
+            authorize_account_update.restrictions = shall_not_restrictions;
```

Comment:
This highlights another conceptual weakness: if a new extension is added to the protocol, it will per default not be covered by the restrictions, i. e. an existing custom authority could use the new extension in whatever way it likes.

Reply:
With the current and only operation to update a voting slate, this is true.

If the explicit voting operation from BSIP47 is approved and implemented, the authorization to update an account's vote could be more tightly restricted.

Reply:
For voting you are correct, but I was thinking about the problem in a more general scope.

We should document BSIP-40 limitations and caveats somewhere.